cfme/tests/cloud_infra_common/test_html5_vm_console.py

```diff
@@ -0,0 +1,168 @@
+        console_type='VNC',
+    )
+    settings_pg.update()
+
+
+def _configure_server_roles_for_test():
+    """
+    Enable websocket role if it is disabled.
+
+    Currently the fixture cfme/fixtures/base.py,
+    disables the websocket role to avoid intrusive popups.
+    """
+    # TODO:Find correct place to put this method at.
+    roles = configuration.get_server_roles()
+    if 'websocket' in roles and not roles['websocket']:
+        logger.info('Enabling the websocket role to allow console connections')
+        roles['websocket'] = True
+        configuration.set_server_roles(**roles)
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.8', reason='Only valid for >= 5.8')
+def test_html5_vm_console(appliance, provider, vm_obj):
+    """
+    Test the HTML5 console support for a particular provider.
+
+    The supported providers are:
+
+        VMware
+        Openstack
+        RHV
+
+    For a given provider, and a given VM, the console will be opened, and then:
+
+        - The console's status will be checked.
+        - A command that creates a file will be sent through the console.
+        - Using ssh we will check that the command worked (i.e. that the file
+          was created.
+    """
+    # temporarily getting credentials from vspherehost-nested because they
+    # match with the PhotonOS Credentials, but need to add credentials for
+    # templates in the credentials.eyaml
+    photon_os_username = credentials['vspherehost-nested'].get('username')
+    photon_os_password = credentials['vspherehost-nested'].get('password')
+    _configure_server_roles_for_test()  # make sure websocket is enabled
+    if provider.one_of(VMwareProvider):
+        _configure_vmware_console_for_test(appliance)
+
+    vm_obj.open_console(console='VM Console')
+    assert vm_obj.vm_console is not None, 'VMConsole object should be created'
+
+    vm_console = vm_obj.vm_console
+
+    # If the banner/connection-status element exists we can get
+    # the connection status text and if the console is healthy, it should connect.
+    assert vm_console.wait_for_connect() is True, "VM Console did not reach 'connected' state"
+
+    # Get the login screen image, and make sure it is a jpeg file:
+    screen = vm_console.get_screen()
+    assert imghdr.what('', screen) == 'jpeg'
+
+    # Wait is needed for Infra VMs which is not required for Openstack Instances
+    # Since infra VMs take more time to boot up
+    if not provider.one_of(OpenStackProvider):
+        logger.info("Wait 60 sec for 'login:' prompt")
+        time.sleep(60)  # wait for "login:" prompt
+
+    # Try to get the login screen text. We should have the text
+    # login there:
+    console_text = vm_console.get_screen_text()
+    assert re.search("login:", console_text)
+
+    # Enter Username:
+    # XXX: This is hard coded, we need to add some new yaml to track the
+    #      template credentials.
+    if provider.one_of(OpenStackProvider):
```

Comment:
Yes new yaml needs adding before we merge this, add a new substanza on the provider